File: Back-end/src/main/java/com/example/springsilver/models/Product.java
package com.example.springsilver.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
@Table(name = "Product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Long productId;
    private Long quantity;
    //Dodadeno name na productot
    //TODO Izmeni ja bazata
    private String name;

    private String description;
    private float price;

    //Mozebi treba column name ovde?
    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @JoinColumn(name = "image_pathurl")
    private String imagePathURL;

    public Product(Long quantity, String name, String description, float price, Category category, String imagePathURL) {
        this.quantity = quantity;
        this.name = name;
        this.description = description;
        this.price = price;
        this.category = category;
        this.imagePathURL = imagePathURL;
    }
}

File: Front-end/react-library/src/models/product.ts
import CategoryModel from "./category";

class ProductModel {
    productId: number;
    quantity: number;
    name?: string;
    description?: string;
    price?: number;
    imageUrl: string;
    category: CategoryModel | null; // CategoryModel or null if category is optional

    constructor(productId: number, quantity: number, name: string, description: string, price: number, imageUrl: string, category: CategoryModel | null) {
        this.productId = productId;
        this.quantity = quantity;
        this.name = name;
        this.description = description;
        this.price = price;
        this.imageUrl = imageUrl;
        this.category = category;
    }
}

export default ProductModel;
